src/util/taproot.rs

```diff
@@ -760,16 +764,16 @@ impl ControlBlock {
 /// The leaf version for tapleafs
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct LeafVersion(u8);
+pub enum LeafVersion {
+    /// BIP-342 tapscript
+    TapScript,
 
-impl Default for LeafVersion {
-    fn default() -> Self {
-        LeafVersion(TAPROOT_LEAF_TAPSCRIPT)
-    }
+    /// Future leaf version
+    Future(u8)
```

Comment:
This is essentially a public field that allows creating `LeafVersion::Future(TAPROOT_LEAF_TAPSCRIPT)` which could cause breakages/confusion.

I suggest:

```rust
#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
pub struct FutureLeafVersion(u8);

// IMPORTANT: NO PUBLIC CONSTRUCTOR!
// The only way to construct this is by converting u8 to LeafVersion and then extracting it

impl FutureLeafVersion> {
    pub fn to_consensus(self) -> u8 {
        self.0
    }
}

// How to represent it though?
impl Display for FutureLeafVersion { ... }
```

Maybe also perform conversion from consensus encoding to cardinal representation so that we can use `NonZeroUsize`